src/commonMain/kotlin/character/classes/priest/talents/ShadowWeaving.kt

```diff
@@ -0,0 +1,48 @@
+package character.classes.priest.talents
+
+import character.*
+import character.classes.priest.debuffs.ShadowWeavingDebuff
+import data.Constants
+import data.model.Item
+import sim.Event
+import sim.EventResult
+import sim.EventType
+import sim.SimParticipant
+
+import kotlin.random.Random
+
+class ShadowWeaving(currentRank: Int) : Talent(currentRank) {
+    companion object {
+        const val name = "Shadow Weaving"
+    }
+    override val name: String = Companion.name
+    override val maxRank: Int = 5
+
+    val buff = object : Buff() {
+        override val name: String = Companion.name
+        override val durationMs: Int = -1
+        override val hidden: Boolean = true
+
+        val proc = object : Proc() {
+            override val triggers: List<Trigger> = listOf(
+                Trigger.SHADOW_DAMAGE_NON_PERIODIC
+            )
+            override val type: Type = Type.STATIC            
+
+            override fun proc(sp: SimParticipant, items: List<Item>?, ability: Ability?, event: Event?) {
+                if(event?.result != EventResult.RESIST) {
+                    val debuff = sp.sim.target.debuffs.get(Companion.name) ?: ShadowWeavingDebuff(sp)
```

Comment:
Instead of referencing this in abilities, this could be implemented as a stacking buff that gets added to the raid.  This would help avoid issues with double-counting the raid version of this buff.  

Something along the lines of:

```
val swBuff = object : Buff() {
    override val name: String = "Shadow Weaving"
    override val durationMs: Int = 15000
    override val maxStacks: Int = 5

    override fun modifyStats(sp: SimParticipant): Stats {
        return Stats(shadowDamageMultiplier = 1.1)
    }
}
```

And later in `proc()`:

`sp.sim.addRaidBuff(swBuff())`